amy/fiscal/tests/test_filters.py

```diff
@@ -0,0 +1,454 @@
+                # Inhouse
+                + F("inhouse_instructor_training_seats")
+                + F("additional_inhouse_instructor_training_seats")
+                + Coalesce("inhouse_instructor_training_seats_rolled_from_previous", 0)
+                - Count(
+                    "task",
+                    filter=Q(task__role__name="learner", task__seat_public=False),
+                )
+                - Coalesce("inhouse_instructor_training_seats_rolled_over", 0)
+            ),
+        )
+
+        # create 2 used seats on test membership
+        # will make remaining seats negative
+        cls.event = Event.objects.create(
+            host=cls.organization,
+            sponsor=cls.organization,
+            membership=cls.membership,
+            slug="2023-02-16-ttt-test",
+        )
+        cls.person = Person.objects.create(personal="Test", username="test")
+        cls.person2 = Person.objects.create(personal="Test2", username="test2")
+        cls.role = Role.objects.create(name="learner", verbose_name="Learner")
+        cls.task = Task.objects.create(
+            event=cls.event,
+            person=cls.person,
+            role=cls.role,
+            seat_membership=cls.membership,
+            seat_public=True,
+        )
+        cls.task2 = Task.objects.create(
+            event=cls.event,
+            person=cls.person2,
+            role=cls.role,
+            seat_membership=cls.membership,
+            seat_public=True,
+        )
+
+        # get filterset
+        cls.filterset = MembershipFilter({})
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        # clean up used seats
+        cls.task.delete()
+        cls.task2.delete()
+        cls.event.delete()
+        cls.person.delete()
+        cls.person2.delete()
+        cls.role.delete()
+
+        # clean up memberships
+        cls.member.delete()
+        cls.membership.delete()
+        cls.organization.delete()
+        cls.member2.delete()
+        cls.membership2.delete()
+        cls.organization2.delete()
+
+    def test_fields(self):
+        # Arrange & Act stages happen in setUpClass()
+        # Assert
+        self.assertEqual(
+            set(self.filterset.filters.keys()),
+            {
+                "organization_name",
+                "consortium",
+                "public_status",
+                "variant",
+                "contribution_type",
+                "active_only",
+                "training_seats_only",
+                "nonpositive_remaining_seats_only",
+                "order_by",
+            },
+        )
+
+    def test_filter_active_only(self):
+        # Arrange
+        name = "active_only"
+        value = True
+
+        # Act
+        result = self.filterset.filters[name].filter(self.qs, value)
+
+        # Assert
+        self.assertTrue(self.membership in result)
+        self.assertFalse(self.membership2 in result)
```

Comment:
There's also `assertIn` and `assertNotIn`, they may provide better insight if the condition fails.